coderd/parameters.go

```diff
@@ -144,15 +145,17 @@ func (api *API) handleDynamicParameters(listen bool, rw http.ResponseWriter, r *
 	}
 
 	// Add the file first. Calling `Release` if it fails is a no-op, so this is safe.
-	templateFS, err := api.FileCache.Acquire(fileCtx, fileID)
+	var templateFS fs.FS
+	closeableTemplateFS, err := api.FileCache.Acquire(fileCtx, fileID)
 	if err != nil {
 		httpapi.Write(ctx, rw, http.StatusNotFound, codersdk.Response{
 			Message: "Internal error fetching template version Terraform.",
 			Detail:  err.Error(),
 		})
 		return
 	}
-	defer api.FileCache.Release(fileID)
+	defer closeableTemplateFS.Close()
+	templateFS = closeableTemplateFS
```

Comment:
oh I really do not like this. I can see why you're doing it but it feels bad. this definitely at least needs a comment about why this is so awkward.